### ISSUE TITLE
LNCC Training: Unable to learn (very unstable)

### ISSUE THREAD

**Issue by JannikLa** [voxelmorph/voxelmorph#527]
**Task** (what are you trying to do/register?)

I am registering two T1 3D images from the OASIS data set (L2Reg challenge).
The scans are preprocessed (skull-stripped, aligned, normalized).
I am taking a fixed image from one subject and randomly select another subject as moving image (examples below).

_I invested a lot if time in figuring out what is wrong. Maybe someone can give hints on what I could try or what might be wrong in my setup. Any help is appreciated!_

**What have you tried**

I am using the pytroch implemention of voxelmorph. A dataloader loads the inputs to the network.
- The network works as expected with MSE
- With LNCC the loss is very unstable an does not decrease

**Details of experiments**

- My "basline" setup is: lr: 1e-4, bs: 4, Reg weight: 0.1, loss: LNCC (win size 9)
  - I changed hyper-parameters separately in both directions (e.g. increasing and decreasing the learning rate)
  - I also tried setting the Regularization weight to zero
  - If I set the initialization of the flow layer a lot higher (to 1 or 10) the **network is able to overfit**
    - ` self.flow.weight = nn.Parameter(Normal(0, 1e-5).sample(self.flow.weight.shape))` (networks.py, l 214)
 - I also tried debugging the LNCC loss. It works as expected (e.g. tried computing the loss between identical images)

## Image of training loss with maxed out smoothing in visualization ("baseline" setup from above)
![image](https://github.com/voxelmorph/voxelmorph/assets/63288158/98acb236-30e6-4e0b-8da2-5eeb286bd634)

## Image of training loss (LNCC) when overfitting on a pair of images with large initialization weights (Normal(0, 10))
![image](https://github.com/voxelmorph/voxelmorph/assets/63288158/fec49b49-badd-4b45-8d2d-d201c1bc1696)

## Image of input data

![image](https://github.com/voxelmorph/voxelmorph/assets/63288158/bd32cf80-f2fd-4e85-807b-3fa8abd8e2e8)

![image](https://github.com/voxelmorph/voxelmorph/assets/63288158/b1cd3643-9be3-465b-85d1-9c46d888eae9)

**Comment by adalca** [comment]
I haven't worked with the pytorch version that much especially recently, but just so I know -- when you say the MSE version worked, what were the hyperparameters that you used?

What do you mean by  this line -- what does 'overfit' mean here?
> If I set the initialization of the flow layer a lot higher (to 1 or 10) the network is able to overfit
